src/main.cpp

```diff
@@ -2735,13 +2762,23 @@ bool ConnectBlock(const CBlock& block, CValidationState& state, CBlockIndex* pin
         pos.nTxOffset += ::GetSerializeSize(tx, SER_DISK, CLIENT_VERSION);
     }
 
-    assert(trieCache.incrementBlock(blockundo.insertUndo, blockundo.expireUndo, blockundo.insertSupportUndo, blockundo.expireSupportUndo, blockundo.takeoverHeightUndo));
+    if (pindex->nHeight == Params().GetConsensus().nNormalizedNameForkHeight) {
```

Comment:
Is there an advantage to having this code to enable the normalization go here rather than in the CClaimTrieCache object (since the latter also gets IncrementBlock called)? What rule would guide that?